### DISCUSSION TITLE
Which Serialbattery-Adapter

### DISCUSSION THREAD

**Discussion by Miro-2000** [Louisvdw/dbus-serialbattery#949] [Q&A] [answered]
I found a fork of this serialbattery adapter from Mr. Manuel . 

What is the different of this versions ?
v1.0.20230531 (latest) <=> v1.1.20240121 (latest)

Which driver is recommend ?

Thank you 
Werner

**Comment by mr-manuel** [comment]
Just read the CHANGELOG. I always recommend the latest version.

**Reply by Miro-2000** [reply]
Thank you ... , i have installed your driver v1.1.20240121 (latest) succesfully.

Another question :
Can I change the voltage./temperature-steps and add more steps , or are there maximum step restrictions?

Example for temperature steps
default =>
TEMPERATURE_LIMITS_WHILE_CHARGING = 0,   2,   5,  10,  15, 20, 35,  40, 5              
MAX_CHARGE_CURRENT_T_FRACTION      = 0, 0.1, 0.2, 0.4, 0.8,  1,  1, 0.4,  0

new => added more Steps:
TEMPERATURE_LIMITS_WHILE_CHARGING = 0, 3,    5,   7,  10,  12,  15, 20, 35,  40,  45,  50, 55
MAX_CHARGE_CURRENT_T_FRACTION       = 0, 0, 0.15, 0.4, 0.6, 0.7, 0.8,  1,  1, 0.9, 0.8, 0.3,  0

Thank you

**Reply by mr-manuel** [reply, marked as answer]
You can add as many steps as you want. The more steps you add, the more CPU is used. Track your system stability to be sure all is working correctly and the system is not crashing and rebooting.

**Reply by ramack** [reply]
This is interesting. I would never have expected that the number of steps plays a role. Do you have empiric data for this increase of CPU load? - in my system.with 5 BMSs I always had trouble with CPU load and decreased the pollig time for th BMS reading to make it somehow stable but if I could gain performance by changing such configurations it would be cool.

**Reply by mr-manuel** [reply]
You have to test, I do not have more details.
